segmentation/utils/spatial_analysis_utils.py

```diff
@@ -4,38 +4,86 @@
 import skimage.measure
 import scipy
 import statsmodels
+import h5py
 from statsmodels.stats.multitest import multipletests
 from scipy.spatial.distance import cdist
 
 
-def calc_dist_matrix(label_map):
+def calc_dist_matrix(label_map, ret=False):
```

Comment:
We don't to just save this in whatever directory the code is being run in, we should add a required argument so specify a path to a folder